### DISCUSSION TITLE
Help with regex parsing package.json

### DISCUSSION THREAD

**Discussion by jmctune** [renovatebot/renovate#20829] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

Never worked.

### Wanted end result.

Trying to update a `package.json` file that has the following contents:

```json
{
  "version": "0.1.0",
  "private": true,
  "engines": {
    "node": "18.14.0"
  }
}
```

I want to update `18.14.0` -> `18.14.2`, which is an available version in my configured datasource. I don't want to use the `npm` manager to accomplish this as I need to update this version of node to a specific version and not what's latest.

### What you tried so far.

I have the following regex rule written:

```json
{
  "regexManagers": [
    {
      "fileMatch": ["^package.json$"],
      "matchStrings": [
        "\"node\":\\s\"(?<currentValue>[0-9\\.]*)\""
      ],
      "depNameTemplate": "xxx.dkr.ecr.us-east-1.amazonaws.com/node",
      "datasourceTemplate": "docker",
      "autoReplaceStringTemplate": "{{{newValue}}}"
    }
  ]
}
```

I've tested that the `<captureGroup>` is successfully assigning itself just to the version using [regex101](https://regex101.com/r/GG52iv/1), but renovate is failing to extract the text from the file. I'm not sure what I'm doing wrong here.

### Relevant debug logs

<details><summary>Logs</summary>

```
...
      {
        "deps": [
          {
            "depName": "xxx.dkr.ecr.us-east-1.amazonaws.com/node",
            "currentValue": "18.14.0",
            "datasource": "docker",
            "replaceString": "\"node\": \"18.14.0\"",
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "18.14.2",
                "newValue": "18.14.2",
                "newMajor": 18,
                "newMinor": 14,
                "updateType": "patch",
                "branchName": "renovate/xxx.dkr.ecr.us-east-1.amazonaws.com-node-18.x"
              }
            ],
            "warnings": [],
            "versioning": "docker",
            "registryUrl": "https://xxx.dkr.ecr.us-east-1.amazonaws.com",
            "currentVersion": "18.14.0",
            "isSingleVersion": true,
            "fixedVersion": "18.14.0"
          }
        ],
        "matchStrings": [
          "\"node\":\\s\"(?<currentValue>[0-9\\.]*)\""
        ],
        "depNameTemplate": "xxx.dkr.ecr.us-east-1.amazonaws.com/node",
        "datasourceTemplate": "docker",
        "autoReplaceStringTemplate": "{{{newValue}}}",
        "packageFile": "package.json"
      }
...
DEBUG: Starting search at index 62(packageFile="package.json", branch="renovate/xxx.dkr.ecr.us-east-1.amazonaws.com-node-18.x")
{
  "depName": "xxx.dkr.ecr.us-east-1.amazonaws.com/node"
}
DEBUG: Found match at index 62(packageFile="package.json", branch="renovate/xxx.dkr.ecr.us-east-1.amazonaws.com-node-18.x")
{
  "depName": "xxx.dkr.ecr.us-east-1.amazonaws.com/node"
}
DEBUG: Could not extract package.json(branch="renovate/xxx.dkr.ecr.us-east-1.amazonaws.com-node-18.x")
WARN: Error updating branch: update failure(branch="renovate/xxx.dkr.ecr.us-east-1.amazonaws.com-node-18.x")
```

</details>

**Comment by viceice** [comment]
why are you using regex? npm manager should be able to handle it native

**Reply by jmctune** [reply]
My problem with the npm manager is that it queries public upstream for the latest version. We build our own node images and publish them to our private ECR repository. Other parts of my renovate config bump the version that's in our ECR, including an nvmrc and Dockerfile. I need them all to bump at the same time and be put into the same PR.

I'm a pretty new user to renovate, but it doesn't seem like npm manager was all that configurable outside of just enabling it. I've been having trouble getting around the documentation.

**Comment by jmctune** [comment, marked as answer]
I've been at this for hours, but had a coworker help me figure it out.

In the logs I posted, there's this line:

```
"replaceString": "\"node\": \"18.14.0\""
```

I needed to write my `autoReplaceStringTemplate` as so:

```
"autoReplaceStringTemplate": "\"node\": \"{{{newValue}}}\""
```

It's now bumping the version correctly.

If there's an easier way to do this with existing renovate functionality, I'm interested. I saw there was an open feature to add json/yaml manager support, which would have made this much less painful.

**Reply by viceice** [reply]
remove the auto-replace template and it should work too
